src/main/routing/shd-router-queue-codel.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * shd-router-queue-codel.c
+ *
+ *  An active queue management (AQM) algorithm implementing CoDel.
+ *  https://tools.ietf.org/html/rfc8289
+ *
+ *  The "Flow Queue" variant is not implemented.
+ *  https://tools.ietf.org/html/rfc8290
+ *
+ *  More info:
+ *   - https://en.wikipedia.org/wiki/CoDel
+ *   - http://man7.org/linux/man-pages/man8/tc-codel.8.html
+ *   - https://queue.acm.org/detail.cfm?id=2209336
+ *   - https://queue.acm.org/appendices/codel.html
+ *
+ *  Created on: Jan 9, 2020
+ *      Author: rjansen
+ */
+
+#include <math.h>
+
+#include "shadow.h"
+
+/* hard limit of queue size, in number of packets */
+//#define CODEL_PARAM_LIMIT 1000
+#define CODEL_PARAM_LIMIT G_MAXUINT
+
+/* target minimum standing queue delay time (5 milliseconds) */
```

Comment:
Delete "5 milliseconds", since the ground truth of the value can be seen below and is already out of date? :)